File: C++/String/14. Longest Common Prefix.cpp
#include <vector>

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int index = 0;
        bool check = true;
        while (check && index < strs[0].size()) {
            char c = strs[0][index];
            for (int i = 1; i < strs.size(); i++) {
                check = index < strs[i].size() && strs[i][index] == c;
                if (!check) break;
            }
            if (check) index++;
        }
        return strs[0].substr(0, index);
    }
};

File: Python/Archive/14. Longest Common Prefix.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        index, check = 0, True
        while check and index < len(strs[0]):
            c = strs[0][index]
            for i in range(1, len(strs)):
                check = index < len(strs[i]) and strs[i][index] == c
                if not check: break
            if check: index += 1
        return strs[0][:index]